lib/Bio/EnsEMBL/DataCheck/Checks/CheckTableSizes.pm

```diff
@@ -0,0 +1,84 @@
+=head1 LICENSE
+
+Copyright [2018-2019] EMBL-European Bioinformatics Institute
+
+Licensed under the Apache License, Version 2.0 (the 'License');
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an 'AS IS' BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+=cut
+
+package Bio::EnsEMBL::DataCheck::Checks::CheckTableSizes;
+
+use warnings;
+use strict;
+
+use Moose;
+use Test::More;
+use Bio::EnsEMBL::Registry;
+use Bio::EnsEMBL::Utils::ConfigRegistry;
+use Bio::EnsEMBL::DBSQL::DBAdaptor;
+use Bio::EnsEMBL::Compara::DBSQL::DBAdaptor;
+use Bio::EnsEMBL::DataCheck::Test::DataCheck;
+use Bio::EnsEMBL::DataCheck::Utils qw/ array_diff /;
+
+extends 'Bio::EnsEMBL::DataCheck::DbCheck';
+
+use constant {
+  NAME           => 'CheckTableSizes',
+  DESCRIPTION    => 'Tables must be populated and not differ significantly in row numbers',
+  GROUPS         => ['compara', 'compara_tables'],
+  DATACHECK_TYPE => 'advisory',
+  DB_TYPES       => ['compara'],
+  TABLES         => []
```

Comment:
I left this empty with the idea that this DC could be used in more ways, for example during production, before all tables have been written and populated, and use on ehive dbs. With this in mind, any issues can be picked up at the end of a pipeline and caught quickly ahead of the end of production just before handover. Do you think this may not be such a good idea? If I put every compara table in the list of TABLES, would the DC still be able to be used during production on pipeline dbs and only partially merged release dbs (for example if we merge just the genomic data and are still waiting on the protein tree data)?

Reply:
As far as I understand `TABLES` is only used in the context of a DC setup that records the status of the tables whenever it is run, and skips tests that touch tables (as declared in `TABLES`) that haven't changed. So we should be able to run the DCs on any database (production, merged, etc) regardless of `TABLES`.